### PR TITLE
First version of a SteamshipVectorStore

### PR DESCRIPTION
**Pull request by EniasCailliau**
> 🚨 This PR depends on https://github.com/steamship-plugins/openai-embedder/pull/1

# This PR

Introduces a LangChain compatible `SteamshipVectorStore`.

This allows people to switch from FAISS-based VectorDB + QA examples -> Steamship just by swapping 
`from langchain.vectorstores.faiss import FAISS` --> `from steamship_langchain.vectorstores.steamship import SteamshipVectorStore` 

I also included code that derives the appropriate embedding model by inspecting the `langchain.embeddings.OpenAIEmbeddings` model. 

A functional example is included under `basic_examples/use_cases/vectorDBQA/after.py`.

### CODE REVIEW

File: examples/qa_with_sources/client/client.py
Comment (douglas-reid) [inline examples/qa_with_sources/client/client.py:34]: just so i follow along properly: this is the result of switching to `VectorDBQAWithSourcesChain` from `load_qa_with_sources_chain` ?

File: examples/qa_with_sources/server/api.py
Comment (douglas-reid) [inline examples/qa_with_sources/server/api.py:30]: i think in the future including metadata that includes the **actual** offsets from the original files will be a potential win.

File: src/steamship_langchain/vectorstores/steamship_vector_store.py
Comment (douglas-reid) [inline src/steamship_langchain/vectorstores/steamship_vector_store.py:68]: i don't think it matters too much, since results should return within 3m, but i'm wondering what guideline we want to establish for wrapping / not wrapping this in `try`.

File: tests/llms/test_openai.py
Comment (douglas-reid) [inline tests/llms/test_openai.py:1]: these tests all seem great. did you mean to include them with this PR?  or should they be a distinct set?

File: docs/support/coverage.md
Comment (douglas-reid) [inline docs/support/coverage.md:11]: curious: it seems like there is nothing preventing this, necessarily. what is left to be done?

File: docs/use_cases/question_answering.md
Comment (douglas-reid) [inline docs/use_cases/question_answering.md:4]:
```suggestion
> When you want to use a stateful database it is recommended to use the `SteamshipVectorStore` **
```

**Review by douglas-reid — COMMENTED** [review]
LG. Just a few questions to make sure I understand fully.

This must be sequenced after the other PR, correct?

File: examples/qa_with_sources/server/steamship.json
Comment (dkolas) [inline examples/qa_with_sources/server/steamship.json:3]: Did you mean to take this back out?

**Review by dkolas — COMMENTED** [review]
Looks good to me, though I don't know enough LC yet to totally get it.  One note about package handle.

File: examples/qa_with_sources/client/client.py
Comment (EniasCailliau) [reply examples/qa_with_sources/client/client.py:34]: Yes! But, I still have to test this out before I can claim this works.

File: src/steamship_langchain/vectorstores/steamship_vector_store.py
Comment (EniasCailliau) [reply src/steamship_langchain/vectorstores/steamship_vector_store.py:68]: You mean when the .wait() times out? Good question, maybe in the future we expose the timeout in the vector store interface? That does deviate from LangChain's interface though which makes it less ideal.

File: docs/support/coverage.md
Comment (EniasCailliau) [reply docs/support/coverage.md:11]: See Previous PR. There's a namespacing issue with our OpenAI implementation.

File: examples/qa_with_sources/server/steamship.json
Comment (EniasCailliau) [reply examples/qa_with_sources/server/steamship.json:3]: Yes! 🦅 👀

**Comment by douglas-reid** [comment]
I think this needs a rebase now.

File: README.md
Comment (douglas-reid) [inline README.md:3]: i'm confused by this appearing in the diff, as this doesn't appear on main any more. Any idea what is happening here?

File: examples/qa_with_sources/client/client.py
Comment (douglas-reid) [reply examples/qa_with_sources/client/client.py:40]: i assume this is now verified.

File: src/steamship_langchain/vectorstores/steamship_vector_store.py
Comment (douglas-reid) [inline src/steamship_langchain/vectorstores/steamship_vector_store.py:40]: question: does dimensionality need to be a param ?  is it tied to the `embedding` ?

**Review by douglas-reid — COMMENTED** [review]
Final question.

File: tests/llms/test_openai.py
Comment (douglas-reid) [reply tests/llms/test_openai.py:1]: 👍

File: src/steamship_langchain/vectorstores/steamship_vector_store.py
Comment (douglas-reid) [reply src/steamship_langchain/vectorstores/steamship_vector_store.py:40]: after offline convo, my understanding is that this will now be a param to the `__init__`. will approve to unblock merge with that change in the timezone change-over ;)

**Review by douglas-reid — APPROVED** [review]
Approved. Waiting on the small UX patch for embedding + dimensionality, but that doesn't need another review.

**Pull request merged**
